frontend/src/app/gitImport.tsx

```diff
@@ -0,0 +1,159 @@
+                    action={<AlertActionCloseButton onClose={this.dismissNotification} />}
+                  />
+                </div>
+              )}
+              {isFormSubmitted && isSucess && (
+                <div className="notification-container">
+                  <Alert
+                    variant="success"
+                    title={this.state.submitMsg}
+                    action={<AlertActionCloseButton onClose={this.setRedirect} />}
+                  />
+                </div>
+              )}
+             {isFormSubmitted && !isSucess && (
+                <div className="notification-container">
+                  <Alert
+                    variant="danger"
+                    title={this.state.submitMsg}
+                    action={<AlertActionCloseButton onClose={this.dismissNotification} />}
+                  />
+                </div>
+              )}
+
+              <FormGroup
+                label="Repository URL"
+                fieldId="repository-url"
+              >
+                <TextInput id="repository-url" type="text" placeholder="Repository" value={repository} onChange={this.handleRepoInput} />
+              </FormGroup>
+              <br />
+              <FormGroup
+                label="Branch"
+                fieldId="branch-name"
+              >
+                <TextInput id="branch-name" type="text" placeholder="Branch" value={branch} onChange={this.handleBranchInput} />
+              </FormGroup>
+              <br />
+              <Button aria-label="Submit the repository and branch information to the git integration service." onClick={this.cloneRepo}>Submit</Button>
+              <div>
+                {this.loginRedirect()}
+                {this.renderRedirect()}
+              </div>
+            </div>
+          </div>
+        </Bullseye>
+      </React.Fragment>
+    );
+  }
+
+  private handleRepoInput = repository => {
+    this.setState({ repository });
+    console.log("Repository " + repository)
+  };
+
+  private handleBranchInput = branch => {
+    this.setState({ branch });
+    console.log("Branch " + branch)
+  };
+
+  private cloneRepo = (postBody) => {
+    console.log("My repo is: " + this.state.repository + " and branch is " + this.state.branch + ".")
+    if (this.state.repository === "") {
+      this.setState({ isMissingFields: true })
+    } else {
+      const hdrs = {
+        'Accept': 'application/json',
+        'cache-control': 'no-cache'
+      }
+
+      fetch('/content/repositories/GIT_INTEGRATION_URL')
+        .then((resp) => resp.text()).then((text) => {
+          this.setState({ git2pantheonURL: text })
+        }).then(() => {
+          console.log("The git2pantheon URL is: " + this.state.git2pantheonURL)
+          const payload = {
+            branch: this.state.branch,
+            repo: this.state.repository
+          };
+          fetch(this.state.git2pantheonURL + "/clone", {
```

Comment:
Although I'm not in love with this double fetch, I think I understand why you have to do it. There's no way for pantheon to dynamically bake the git url into the javascript, so this is the next best option, right?

Reply:
yeah the other way would be using build time ENV variables. I think we are stuck with this for now.